### ISSUE TITLE
Relatedness calculation

### ISSUE THREAD

**Issue by tanghaibao** [brentp/cyvcf2#122]
@brentp 

Recently I needed a way to calculate the genetic relatedness, and was going through the Peddy formula on calculation of genetic relatedness, and saw that the defined formula:

![image](https://user-images.githubusercontent.com/106987/61733605-06c5a980-ad35-11e9-82e3-e4abf4173a1c.png)

Going through the actual calculation in the code, I found the bottom part of the formula was modified to a `min` of the two hets:

https://github.com/brentp/cyvcf2/blob/master/cyvcf2/cyvcf2.pyx#L873

<img width="924" alt="image" src="https://user-images.githubusercontent.com/106987/61733762-5c9a5180-ad35-11e9-8cdb-2a583b7661d3.png">

I was wondering if there is a reason for this change, or perhaps I wasn't looking in the right place?

Thanks,
Haibao

**Comment by brentp** [comment]
hey Haibao,
that seems to work better for me to dampen the worst cases. I also use min [in somalier](https://github.com/brentp/somalier/blob/master/src/somalierpkg/relate.nim#L101)

this seems to matter much less that site-selection, in my experience. For that reason, I am happy to adjust if you think it worth while.

**Comment by tanghaibao** [comment]
Hi Brent,

Thanks for the fast response. I was more interested in empirically matching the expected relatedness in real pedigrees. I wonder if you have ever done similar plots to compare with geometric mean, simple average, min, max and compute correlations, like the figures you showed in Peddy documentation (nice).

I tend to think that this matters less too, but would like to have some numbers behind the decision (or perhaps they are all similar with one another). I am fine either way but am just curious.

Haibao

**Comment by brentp** [comment]
Yes, I looked at this pretty extensively. On CEPH where we have 603 samples and many multi-generational pedigrees with known relationships, min seemed to do the best. I don't recall what my definition of best was, but I revisited this for somalier and came to the same conclusion. 

It also made intuitive sense because the proportion of shared hets (used in calculating relatedness) would be bound by the minimal number of hets between 2 samples.
